File: myskills/frontend/src/services/skill.ts
import api from './api';

interface SkillConfig {
  function_calling: {
    description: string;
    parameters: {
      type: string;
      properties: {
        query: {
          type: string;
          description: string;
        };
        knowledge_base_id: {
          type: string;
          description: string;
        };
      };
      required: string[];
    };
  };
  retrieval: {
    top_k: number;
    similarity_threshold: number;
  };
}

interface SkillMetadata {
  name: string;
  description: string;
  parameters: {
    type: string;
    properties: {
      query: {
        type: string;
        description: string;
      };
      knowledge_base_id: {
        type: string;
        description: string;
      };
    };
    required: string[];
  };
}

interface SkillTestResult {
  query: string;
  results: Array<{
    content: string;
    similarity: number;
  }>;
}

class SkillService {
  async getSkillConfig(): Promise<SkillConfig> {
    return api.get('/skill/config');
  }

  async updateSkillConfig(config: Partial<SkillConfig>): Promise<{ message: string }> {
    return api.put('/skill/config', config);
  }

  async getSkillMetadata(): Promise<SkillMetadata> {
    return api.get('/skill/metadata');
  }

  async testSkill(query: string, kbId?: string): Promise<SkillTestResult> {
    return api.post('/skill/test', { query, knowledge_base_id: kbId });
  }
}

export default new SkillService();

File: myskills/backend/app/routers/skill.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.database import get_db

router = APIRouter()

@router.get("/config", response_model=dict)
async def get_skill_config(db: Session = Depends(get_db)):
    """获取Skill配置"""
    # 这里只是一个占位实现，实际配置需要从数据库或配置文件中获取
    return {
        "function_calling": {
            "description": "MySkills私有文档Skill",
            "parameters": {
                "type": "object",
                "properties": {
                    "query": {
                        "type": "string",
                        "description": "用户的问题"
                    },
                    "knowledge_base_id": {
                        "type": "string",
                        "description": "知识库ID"
                    }
                },
                "required": ["query"]
            }
        },
        "retrieval": {
            "top_k": 5,
            "similarity_threshold": 0.7
        }
    }

@router.put("/config", response_model=dict)
async def update_skill_config(config_data: dict, db: Session = Depends(get_db)):
    """更新Skill配置"""
    # 这里只是一个占位实现，实际更新逻辑需要在服务层实现
    return {"message": "Skill配置更新成功"}

@router.get("/metadata", response_model=dict)
async def get_skill_metadata(db: Session = Depends(get_db)):
    """获取Function Calling元数据"""
    return {
        "name": "myskills_retrieve",
        "description": "从MySkills私有文档中检索相关信息",
        "parameters": {
            "type": "object",
            "properties": {
                "query": {
                    "type": "string",
                    "description": "用户的问题"
                },
                "knowledge_base_id": {
                    "type": "string",
                    "description": "知识库ID"
                }
            },
            "required": ["query"]
        }
    }

@router.post("/test", response_model=dict)
async def test_skill(test_data: dict, db: Session = Depends(get_db)):
    """测试Skill调用"""
    # 这里只是一个占位实现，实际测试逻辑需要在服务层实现
    return {
        "query": test_data.get("query"),
        "results": [
            {
                "content": "测试检索结果",
                "similarity": 0.95
            }
        ]
    }
